File: CodeWars/7kyu/JavaScript/001-isograms.js
// https://www.codewars.com/kata/54ba84be607a92aa900000f1
// An isogram is a word that has no repeating letters, consecutive or non-consecutive. 
// Implement a function that determines whether a string that contains only letters is an isogram. 
// Assume the empty string is an isogram. Ignore letter case.

function isIsogram(str) {
    // convert to lowercase
    const string = str.toLowerCase();

    // empty string is an isogram'
    if (string == "") return true;

    // split into single chars the string
    let chars = string.split("");

    // create empty array for single chars
    let uniques = [];

    // loop through chars and if to unique if not available yet
    for (let char of chars) {
        if (uniques.indexOf(char) == -1) {
            uniques.push(char);
        }
    }

    if (chars.length == uniques.length) {
        return true;
    } else {
        return false;
    }
}

console.log('Dermatoglyphics: ' + isIsogram("Dermatoglyphics"));
console.log('isogram: ' + isIsogram("isogram"));
console.log('aba: ' + isIsogram("aba"));
console.log('moOse: ' + isIsogram("moOse"));
console.log('isIsogram: ' + isIsogram("isIsogram"));
console.log(':' + isIsogram(""));

File: CodeWars/7kyu/Python/001-isograms.py
# https://www.codewars.com/kata/54ba84be607a92aa900000f1
# An isogram is a word that has no repeating letters, consecutive or non-consecutive.
# Implement a function that determines whether a string that contains only letters is an isogram.
# Assume the empty string is an isogram. Ignore letter case.

def is_isogram(string):
    letters = set(string.lower())

    return len(letters) == len(string)


print(is_isogram("Dermatoglyphics"), True)
print(is_isogram("isogram"), True)
print(is_isogram("aba"), False, "same chars may not be adjacent")
print(is_isogram("moOse"), False, "same chars may not be same case")
print(is_isogram("isIsogram"), False)
print(is_isogram(""), True, "an empty string is a valid isogram")
